Code/Framework/AzCore/AzCore/base.h

```diff
@@ -138,6 +138,18 @@ namespace AZ::Internal
 #   define azlocaltime                                      localtime_r
 #endif
 
+// When using -ffast-math flag INFs and NaNs are not handled and
+// it is expected that isfinite() have undefined behaviour.
+// In this case we will provide a replacement following IEEE 754 standard.
+#ifdef __FAST_MATH__
+    #undef azisfinite
+    constexpr bool azisfinite(float f) noexcept
+    {
+        union { float f; uint32_t x; } u = { f };
+        return (u.x & 0x7F800000U) != 0x7F800000U;
```

Comment:
I used first a `reinterpret_cast` intead of the union, but then the function couldn't be `constexpr` in that case which I think it's preferred.

Reply:
yeah there's this great talk on type punning that's worth a watch - https://youtu.be/_qzMpk-22cc. Essentially you should try and avoid the union float/int thing and instead use memcpy to copy from the distinct types (we do this in other places though so you can leave as-is for now but it could theoretically break in future unfortunately).